runtime/interpreter/decode.go

```diff
@@ -1736,3 +1627,186 @@ func (d *DecoderV4) decodeCapabilityStaticType() (StaticType, error) {
 		BorrowType: borrowStaticType,
 	}, nil
 }
+
+// decodeCompositeMetaInfo decodes the meta info from the byte content and updates the composite value.
+// Meta info includes:
+//    - location
+//    - qualifiedIdentifier
+//    - kind
+//
+// This also extracts out the fields raw content and cache it separately inside the value.
+//
+func decodeCompositeMetaInfo(v *CompositeValue, content []byte) error {
+	d, err := NewByteDecoder(content, v.Owner, CurrentEncodingVersion, nil)
+	if err != nil {
+		return err
+	}
+
+	const expectedLength = encodedCompositeValueLength
+
+	size, err := d.decoder.DecodeArrayHead()
+
+	if err != nil {
+		if e, ok := err.(*cbor.WrongTypeError); ok {
+			return fmt.Errorf("invalid composite encoding (@ %s): expected [%d]interface{}, got %s",
+				strings.Join(v.valuePath, "."),
+				expectedLength,
+				e.ActualType.String(),
+			)
+		}
+		return err
+	}
+
+	if size != expectedLength {
+		return fmt.Errorf("invalid composite encoding (@ %s): expected [%d]interface{}, got [%d]interface{}",
+			strings.Join(v.valuePath, "."),
+			expectedLength,
+			size,
+		)
+	}
+
+	// Location
+
+	// Decode location at array index encodedCompositeValueLocationFieldKey
+	location, err := d.decodeLocation()
+	if err != nil {
+		return fmt.Errorf(
+			"invalid composite location encoding (@ %s): %w",
+			strings.Join(v.valuePath, "."),
+			err,
+		)
+	}
+
+	// Skip obsolete element at array index encodedCompositeValueTypeIDFieldKey
+	err = d.decoder.Skip()
+	if err != nil {
+		return err
+	}
+
+	// Kind
+
+	// Decode kind at array index encodedCompositeValueKindFieldKey
+	encodedKind, err := d.decoder.DecodeUint64()
+	if err != nil {
+		if e, ok := err.(*cbor.WrongTypeError); ok {
+			return fmt.Errorf(
+				"invalid composite kind encoding (@ %s): %s",
+				strings.Join(v.valuePath, "."),
+				e.ActualType.String(),
+			)
+		}
+		return err
+	}
+
+	kind := common.CompositeKind(encodedKind)
+
+	// Fields
+
+	fieldsContent, err := d.decoder.DecodeRawBytes()
```

Comment:
What do you think about using `d.decoder.DecodeRawBytesZeroCopy` here?

Reply:
Oh, yes, I used it at the other place, but missed out using it here. Good catch!